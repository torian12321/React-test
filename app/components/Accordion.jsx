var
React     = require('react'),
PropTypes = require('prop-types');

class Accordion extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen    : this.props.isOpen,
            wrappClass: this.props.isOpen ? 'accordion active' : 'accordion'
        };
    }
    expand(){
        this.setState({
            isOpen    : !this.state.isOpen,
            wrappClass: !this.state.isOpen ? 'accordion active' : 'accordion'
        });
    }
    render() {
        return (
            <div className={this.state.wrappClass}>
                <div className="acc-header" onClick= { this.expand.bind(this) }>
                    { this.props.header }
                </div>
                <div className="acc-body">
                    <div className="acc-body-content">
                        { this.props.children }
                    </div>
                </div>
            </div>
        );
    }
}

Accordion.propTypes = {
    header  : PropTypes.string.isRequired,
    children: PropTypes.any.isRequired,
    isOpen  : PropTypes.bool.isRequired
};
Accordion.defaultProps = {
    header: 'Show Details',
    isOpen: false
};

module.exports = Accordion;
