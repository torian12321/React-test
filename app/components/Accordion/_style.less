@import '~LESS_conf';

.accordion{
	@color_main: @themeColor_2;
	@color_sec : @themeColor_1;


	margin    : 10px 0;
	width     : 100%;
	overflow  : hidden;
	background: @color_main;
    font-size : @fontSize_base;
	.rounded();
	.transition();
	.smart-text-color(@color_main);
	.acc-header{
		@icon_size: 20px;
		@gutter_x : 10px;

font-size: 20px;
line-height: 2;
		position     : relative;
		width        : 100%;
		padding      : 0 @gutter_x 0 (@gutter_x + @icon_size);
		cursor       : pointer;
		background   : darken(@color_main, 10%);
		border-bottom: darken(@color_main, 10%) 2px solid;
		.transition(background border-color);
		.smart-text-color(darken(@color_main, 10%));
		&:before{
			position  : absolute;
			content   : '+';
			top       : 0;
			left      : @gutter_x / 2;
			width     : @icon_size;
			height    : @icon_size;
			font-size : @icon_size;
			text-align: center;
			//.transition(transform color);
		}
	}
	.acc-body{
		position  : relative;
		max-height: 0;
		.transition(max-height);
		& > .acc-body-content{
			padding: 20px 10px;
		}
	}

	&:hover{
		.shadow();
	}
	&.active{
		background: lighten(@color_sec, 5%);
		.smart-text-color(lighten(@color_sec, 5%));
		.acc-header{
			background  : @color_sec;
			border-color: darken(@color_sec, 5%);
			&:before{
				content: '-'
			}
		}
		.acc-body{
			max-height: 500px;
		}
	}
}