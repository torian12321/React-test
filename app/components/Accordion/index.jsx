var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames');;

class Accordion extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: this.props.isOpen
        };
    }
    componentWillReceiveProps(nextProps) {
        if (nextProps.isOpen !== this.state.isOpen) {
            this.setState({isOpen: nextProps.isOpen});
        }
    }
    
    expand(){
        this.setState({
            isOpen: !this.state.isOpen
        });
    }
    render() {
        return (
            <div className={classNames('accordion', {'active': this.state.isOpen})}>
                <div className="acc-header" onClick= {this.expand.bind(this)}>
                    {this.props.header}
                </div>
                <div className="acc-body">
                    <div className="acc-body-content">
                        {this.props.children}
                    </div>
                </div>
            </div>
        );
    }
}

Accordion.propTypes = {
    header  : PropTypes.string.isRequired,
    children: PropTypes.any.isRequired,
    isOpen  : PropTypes.bool.isRequired
};
Accordion.defaultProps = {
    header: 'Show Details',
    isOpen: false
};

module.exports = Accordion;
