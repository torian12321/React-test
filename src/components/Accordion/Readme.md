#### Basic
```
<Accordion>Content</Accordion>
```

#### Custom title
```
<Accordion title='Custom title'>Content</Accordion>
```

#### Open by default
```
<Accordion isOpen>Content</Accordion>
```
