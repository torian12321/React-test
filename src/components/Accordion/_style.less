@import '~LESS_conf';

@color_main: @themeColor_2;
@color_sec : @themeColor_1;
@icon_w    : 14px;
@icon_h    : 3px;

.accordion{
	margin    : 10px 0;
	width     : 100%;
	overflow  : hidden;
	background: @color_main;
    font-size : @fontSize_base;
	.rounded();
	.transition();
	.smart-text-color(@color_main);
	.acc-header{		
		@gutter_x : 10px;

        font-size    : 1.2em;
        line-height  : 2;
		position     : relative;
		width        : 100%;
		padding      : 0 @gutter_x 0 ((@gutter_x * 2) + @icon_w);
		cursor       : pointer;
		background   : @color_main, 10%;
		border-bottom: @color_main, 10% 2px solid;
		.transition(background border-color);
		.smart-text-color(@color_main);
		&:before,
        &:after{
			content   : '';
            position  : absolute;
			top       :  ~"calc(1em - ((@{icon_h}) / 2))";
			left      : @gutter_x;
			width     : @icon_w;
			height    : @icon_h;
            .smart-bg-color(@color_main);
            .rounded(@icon_h);
			.transition(transform);
		}
        &:before{.rotate(90deg);}
        &:after {.rotate(180deg);}
	}
	.acc-body{
		position  : relative;
		max-height: 0;
		.transition(max-height);
		& > .acc-body-content{
			padding: 20px 10px;
		}
	}

	&:hover{
		.shadow();
	}
	&.active{
		background: lighten(@color_sec, 5%);
		.smart-text-color(lighten(@color_sec, 5%));
		.acc-header{
			background  : @color_sec;
			border-color: darken(@color_sec, 5%);
            &:before,
            &:after{
                .rotate(0);
            }
		}
		.acc-body{
			max-height: 500px;
		}
	}
}
