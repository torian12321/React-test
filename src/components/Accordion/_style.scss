@import '../../styles/conf/index.scss';

// Vars
$accordion-font-size : $fontSize !default;
$accordion-color-main: $themeColor_2 !default;
$accordion-color-sec : $themeColor_1 !default;
$accordion-icon-w    : 14px !default;
$accordion-icon-h    : 3px !default;

// Block
.accordion{
  margin    : 10px 0;
  width     : 100%;
  overflow  : hidden;
  background: $accordion-color-main;
  font-size : $accordion-font-size;
  @include rounded();
  @include transition();
  @include smart-text-color($accordion-color-main);

  // Elements
	&__header {		
		$gutter_x : 10px;

    font-size  : 1.2em;
    line-height: 2;
    position   : relative;
    width      : 100%;
    padding    : 0 $gutter_x 0 (($gutter_x * 2) + $accordion-icon-w);
    cursor     : pointer;

    // @include transition(background border-color);
    @include smart-text-color($accordion-color-main);
    &:before,
    &:after {
      content : '';
      position: absolute;
      top     : calc(1em - (#{$accordion-icon-h}) / 2);
      left    : $gutter_x;
      width   : $accordion-icon-w;
      height  : $accordion-icon-h;

      @include rounded();
      @include transition(transform);
      @include smart-bg-color($accordion-color-main);
		}
    &:before{@include rotate(90deg);}
    &:after {@include rotate(180deg);}
	}
	&__body {
		position  : relative;
		max-height: 0;
		@include transition(max-height);
    
    &__content {
			padding: 20px 10px;
		}
	}

  // Modifiers
  &:hover {
    @include shadow();
  }
  &--active {
    background: lighten($accordion-color-sec, 5%);
    @include smart-text-color(lighten($accordion-color-sec, 5%));

    .accordion {
      &__header {
        background  : $accordion-color-sec;
        border-color: darken($accordion-color-sec, 5%);

        &:before,
        &:after {
          @include rotate(0);
        }
      }
      &__body {
        max-height: 500px;
      }
    }
  }
}