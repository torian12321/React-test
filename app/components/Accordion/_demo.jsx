import React from 'react'
import Demo, {props as P} from 'react-demo'
import Accordion from './index'


module.exports = () => (
	<div>
		<Demo
			target={Accordion}
				props ={{
				children: P.string('Accordion content')
			}}
		/>
		<Demo
			target={Accordion}
			props ={{
				children: P.string('Accordion content'),
				title   : P.string('Title'),
				isOpen  : P.bool(true),
			}}
		/>
	</div>
);