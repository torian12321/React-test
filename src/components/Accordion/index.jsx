import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style';

class Accordion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen    : this.props.isOpen,
      body_style: {}
    };
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.isOpen !== this.state.isOpen) {
      this.setState({isOpen: nextProps.isOpen});
    }
  }
    
  expand(){
    const accContent_h = this.accContent.clientHeight;

    this.setState({
      isOpen    : !this.state.isOpen,
      body_style: {
        maxHeight: !this.state.isOpen ? accContent_h : 0
      }
    });
  }

  render() {
    return (
      <div className={classNames('accordion', {'active': this.state.isOpen})}>
        <div className="acc-header" onClick= {this.expand.bind(this)}>
          {this.props.title}
        </div>
        <div className="acc-body" style={this.state.body_style}>
          <div className="acc-body-content" ref={(c) => { this.accContent = c; }}>
            {this.props.children}
          </div>
        </div>
      </div>
    );
  }
}

Accordion.propTypes = {
  /** Accordion content. */
  children: PropTypes.any.isRequired,
  title   : PropTypes.string,
  isOpen  : PropTypes.bool
};
Accordion.defaultProps = {
  title : 'Show Details',
  isOpen: false
};

module.exports = Accordion;
