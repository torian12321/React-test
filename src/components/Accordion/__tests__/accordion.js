import React     from 'react';
import renderer  from 'react-test-renderer';
import {shallow} from 'enzyme';
import Accordion from '../index';


it('Should render Accordion', () => {
  const component = renderer.create(
    <Accordion>Accordion content</Accordion>
  );
  let tree = component.toJSON();

  expect(component.toJSON()).toMatchSnapshot();
});