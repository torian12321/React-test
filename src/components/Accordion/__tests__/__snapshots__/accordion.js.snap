// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Accordion 1`] = `
<div
  className="accordion"
>
  <div
    className="acc-header"
    onClick={[Function]}
  >
    Show Details
  </div>
  <div
    className="acc-body"
    style={Object {}}
  >
    <div
      className="acc-body-content"
    >
      Accordion content
    </div>
  </div>
</div>
`;
